internal/onepassword/sdk/sdk_client.go

```diff
@@ -0,0 +1,168 @@
+			result = append(result, vault)
+		}
+	}
+
+	return result, nil
+}
+
+func (c *Client) GetItem(ctx context.Context, itemUuid, vaultUuid string) (*model.Item, error) {
+	sdkItem, err := c.sdkClient.Items().Get(ctx, vaultUuid, itemUuid)
+	if err != nil {
+		return nil, err
+	}
+
+	result := conversions.FromSDKItem(&sdkItem)
+	return result, nil
+}
+
+func (c *Client) GetItemByTitle(ctx context.Context, title string, vaultUuid string) (*model.Item, error) {
+	items, err := c.sdkClient.Items().List(ctx, vaultUuid)
+	if err != nil {
+		return nil, err
+	}
+
+	var matchedID string
+	var count int
+
+	for _, item := range items {
+		if item.Title == title {
+			matchedID = item.ID
+			count++
+		}
+	}
+
+	return c.GetItem(ctx, matchedID, vaultUuid)
+}
+
+func (c *Client) CreateItem(ctx context.Context, item *model.Item, vaultUuid string) (*model.Item, error) {
+	params := conversions.ToSDKItem(item, vaultUuid)
+
+	sdkItem, err := c.sdkClient.Items().Create(ctx, params)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create item: %w", err)
+	}
+
+	result := conversions.FromSDKItem(&sdkItem)
+	return result, nil
+}
+
+func (c *Client) UpdateItem(ctx context.Context, item *model.Item, vaultUuid string) (*model.Item, error) {
+	currentItem, err := c.sdkClient.Items().Get(ctx, vaultUuid, item.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	params := conversions.ToSDKItem(item, vaultUuid)
+	currentItem.Title = params.Title
+	currentItem.Category = params.Category
+	currentItem.Fields = params.Fields
+	currentItem.Sections = params.Sections
+	currentItem.Tags = params.Tags
+	currentItem.Websites = params.Websites
+	if params.Notes != nil {
+		currentItem.Notes = *params.Notes
+	}
+
+	updatedItem, err := c.sdkClient.Items().Put(ctx, currentItem)
+	if err != nil {
+		return nil, err
+	}
+
+	// Convert back to provider model
+	result := conversions.FromSDKItem(&updatedItem)
+	return result, nil
+}
+
+func (c *Client) DeleteItem(ctx context.Context, item *model.Item, vaultUuid string) error {
+	err := c.sdkClient.Items().Delete(ctx, vaultUuid, item.ID)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *Client) GetFileContent(ctx context.Context, file *model.ItemFile, itemUUID, vaultUUID string) ([]byte, error) {
+	fileAttributes := opsdk.FileAttributes{
+		Name: file.Name,
+		ID:   file.ID,
+		Size: uint32(file.Size),
+	}
+
+	content, err := c.sdkClient.Items().Files().Read(ctx, vaultUUID, itemUUID, fileAttributes)
+	if err != nil {
+		return nil, err
+	}
+
+	return content, nil
+}
+
+func NewClient(ctx context.Context, providerUserAgent, serviceAccountToken string, accountToken string) (*Client, error) {
+	var sdkClient *opsdk.Client
+	var err error
+
+	// Initialize with service account token if provided, otherwise use desktop integration
+	if serviceAccountToken != "" {
+		sdkClient, err = opsdk.NewClient(ctx,
+			opsdk.WithServiceAccountToken(serviceAccountToken),
+			opsdk.WithIntegrationInfo("terraform-provider", providerUserAgent),
```

Comment:
Currently, we construct user agent value at [the very top](https://github.com/1Password/terraform-provider-onepassword/blob/main/internal/provider/provider.go#L135) and pass it as a config property. We do not need `"terraform-provider"` string hardcoded here, but instead consume the values from `providerUserAgent` parameter.